rust/protocol/src/identity_key.rs

```diff
@@ -51,20 +55,23 @@ impl From<PublicKey> for IdentityKey {
     }
 }
 
+/// Wrapper for [KeyPair].
 #[derive(Copy, Clone)]
 pub struct IdentityKeyPair {
     identity_key: IdentityKey,
     private_key: PrivateKey,
 }
 
 impl IdentityKeyPair {
+    /// Create a new key pair from a public `identity_key` and a private `private_key`.
     pub fn new(identity_key: IdentityKey, private_key: PrivateKey) -> Self {
         Self {
             identity_key,
             private_key,
         }
     }
 
+    /// Generate an unguessable new identity from randomness in `csprng`.
```

Comment:
"unguessable" is probably taking things too far. How about "cryptographically random"?

Reply:
:D probably!!! I wanted to use a strong adjective so that API consumers would be aware that this is where the randomness is injected (even though the trait bound on `R: CryptoRng + Rng` is probably enough to signal that already). I have made this change!